packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -10,13 +10,46 @@ import { SupportError } from './errors/index.mjs'
  * @virtual
  */
 export class GOVUKFrontendComponent {
+  /**
+   * @type {string|undefined}
+   */
+  static moduleName
```

Comment:
**suggestion** Rather than adding an empty static property in JavaScript to solve a TypeScript issue, what if we tried in types by defining a type for the shape we expect for the constructor of a child class:

```js
/**
 * @typedef {typeof GOVUKFrontendComponent} ChildClassConstructor 
 * @property {string} moduleName - The module name that'll be looked for in the DOM when initialising the component
 */
```

Reply:
Wondering if there's not some overlap with the `CompatibleClass` type as well, but maybe not as `createAll` does not necessarily need a child of `GOVUKFrontendComponent` 🤔

Reply:
i think I got this working without `static moduleName` by creating another type and then combining that type and the type of `GOVUKFrontendComponent`